### DISCUSSION TITLE
Why Rust?

### DISCUSSION THREAD

**Discussion by ichxorya** [typst/typst#723] [Q&A] [answered]
I don't know if this question does not sound very smart, but why this project was built in Rust?

**Comment by laurmaedje** [comment, marked as answer]
Because Rust was my favorite language when the project was started (and it is still). It's of course also a good pick for such software because of static guarantees, performance, WebAssembly, etc. but that's not the true reason.

**Reply by kaffarell** [reply]
The true reason is that you gotta "rewrite everything in rust" :)
![image](https://user-images.githubusercontent.com/42062381/231551206-7080881f-8f57-4b59-849f-e24432d1dee1.png)

**Comment by pablodz** [comment]
Performance in Typst is really good thanks due Rust
